source/opt/const_folding_rules.cpp

```diff
@@ -196,7 +254,84 @@ double GetDoubleFromConst(const analysis::Constant* c) {
+                            : static_cast<uint32_t>(fa);                      \
+      std::vector<uint32_t> words = {result};                                 \
+      return const_mgr->GetConstant(result_type, words);                      \
+    } else if (float_type->width() == 64) {                                   \
+      double fa = GetDoubleFromConst(a);                                      \
+      uint32_t result = integer_type->IsSigned()                              \
+                            ? static_cast<uint32_t>(static_cast<int32_t>(fa)) \
+                            : static_cast<uint32_t>(fa);                      \
+      std::vector<uint32_t> words = {result};                                 \
+      return const_mgr->GetConstant(result_type, words);                      \
+    }                                                                         \
+    return nullptr;                                                           \
+  }
+
+// This macro defines a |UnaryScalarFoldingRule| that performs integer to
+// float conversion.
+// TODO(greg-lunarg): Support for 64-bit integer types.
+#define FOLD_ITOF_OP()                                                        \
+  [](const analysis::Type* result_type, const analysis::Constant* a,          \
+     analysis::ConstantManager* const_mgr) -> const analysis::Constant* {     \
+    assert(result_type != nullptr && a != nullptr);                           \
+    const analysis::Integer* integer_type = a->type()->AsInteger();           \
+    const analysis::Float* float_type = result_type->AsFloat();               \
+    assert(float_type != nullptr);                                            \
+    assert(integer_type != nullptr);                                          \
+    if (integer_type->width() != 32)                                          \
+      return nullptr;                                                         \
+    uint32_t ua = GetIntegerFromConst(a);                                     \
+    if (float_type->width() == 32) {                                          \
+      float result_val = integer_type->IsSigned()                             \
+                             ? static_cast<float>(static_cast<int32_t>(ua))   \
+                             : static_cast<float>(ua);                        \
+      spvutils::FloatProxy<float> result(result_val);                         \
+      std::vector<uint32_t> words = {result.data()};                          \
+      return const_mgr->GetConstant(result_type, words);                      \
+    } else if (float_type->width() == 64) {                                   \
+      double result_val = integer_type->IsSigned()                            \
+                              ? static_cast<double>(static_cast<int32_t>(ua)) \
+                              : static_cast<double>(ua);                      \
+      spvutils::FloatProxy<double> result(result_val);                        \
```

Comment:
You can double check, but I believe Alan added something to the constant class to do this.

Reply:
Couldn't find anything in Constant class.